codec-http/src/test/java/io/netty/handler/codec/http/HttpHeadersTest.java

```diff
@@ -76,8 +76,8 @@ public void testAddSelf() {
         headers.add(headers);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testSetSelf() {
+    @Test
```

Comment:
I think it does, and we discussed this recently, but I don't think this violates the API.  I'm still in favor of only throwing an exception if API semantics can't be preserved.